composer/algorithms/weight_standardization/weight_standardization.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import textwrap
+
+import torch
+import torch.nn.utils.parametrize as parametrize
+from torch import nn
+from torch.fx import symbolic_trace
+
+from composer.core import Algorithm, Event, State
+from composer.loggers import Logger
+from composer.utils import module_surgery
+
+log = logging.getLogger(__name__)
+
+__all__ = ['apply_weight_standardization', 'WeightStandardization']
+
+
+def _standardize_weights(W: torch.Tensor):
+    """Function to standardize the input weight ``W``"""
+    reduce_dims = list(range(1, W.dim()))
+    W_var, W_mean = torch.var_mean(W, dim=reduce_dims, keepdim=True, unbiased=False)
+    return (W - W_mean) / (torch.sqrt(W_var + 1e-10))
```

Comment:
What do you think about adding an option here to account for the fan in? Not quite full scaled weight standardization, but a step closer. Default behavior would still be vanilla weight standardization, but scaling by `1/sqrt(fan_in)` here might be helpful, but not different enough to warrant being it's own algorithm.